chore: lint: update linter settings, fix lint errors

rvagg opened the pull request:
Ref: https://github.com/filecoin-project/lotus/issues/11967

This isn't complete, I just banged through a few iterations of it to see the kinds of changes we're being asked to make. I think a better approach here is to add some exclusion error regexes. Most of the issues fall into two categories:

1. Named but unused parameters
2. Acronym capitalisation - Api -> API, Http -> HTTP, Url -> URL, Id -> ID

The latter suck in particular because a lot of them are on API methods. I've started adding explicit exclusions for those in here, but we could opt for a global exclusion (minimal might be something like `var-naming: var [A-Z].+ should be ` to capture public variables).

Or we could just hammer it with exclusions to minimise the diff in this PR in particular, even turn off `var-naming` entirely.

magik6k commented on storage/paths/db_index.go:
```diff
@@ -49,8 +49,8 @@ func NewDBIndex(al *alerting.Alerting, db *harmonydb.DB) *DBIndex {
 	var sectorEntries []struct {
-		StorageId      string
-		MinerId        sql.NullInt64
+		StorageID      string
+		MinerID        sql.NullInt64
```
Are you sure this won't mess with pgx.Select?

rvagg commented:
I think I agree, I was getting slightly angry at it telling me that `HttpServer` _must_ be `HTTPServer`. I'll work on some exclusions, maybe turn off `var-naming` entirely.

ZenGround0 commented:
>  but we could opt for a global exclusion (minimal might be something like var-naming: var [A-Z].+ should be  to capture public variables).

I like the sound of a global exclusion for this

snadrus commented:
I'm alright turning off both of these things. Because these are not actual problems, but style matters

rvagg commented:
I re-did this PR from scratch, and revisited the whole config. Here's the choices I've made:

* Removed the deprecated linters `golint`, `varcheck`, `deadcode` and `scopelint`, replacing with their equivalents `exportloopref` and `revive`
* Removed `golint` specific path exclusions
* Remove `goconst` - it had two separate exclusions that meant it wasn't being used anyway, I think we're experienced enough to not need this
* All message exclusions are more explicit regexes now, they're harder to find ways around (harder to subvert) and it's much clearer in the config the exclusion decisions we're making
* I decided to ditch all exclusions to unchecked errors except for 3 explicit ones in node/repo/fsrepo.go and 2 in paychmgr/simple.go. This mostly required adding `_ =` in place of the explicit exclusions and `logger.SetLogLevel` had a bunch of unchecked but we have far more `_ =` in the repo so I'm just making that consistent.
* A bunch of minor changes that are obvious in the diff
* I decided not to exclude most of the new `empty-block` errors, but instead I ended up rewriting some logic. I get why some people like empty blocks for comments, but I don't believe they ultimately aid in clarity. **Reviewers please pay careful attention to my changes to make sure I didn't change any logic; some are non-trivial.**

magik6k reviewed (commented):
Only looked at the SP side of things, but looks good to me

ZenGround0 commented on chain/checkpoint.go:
```diff
@@ -25,15 +25,15 @@ func (syncer *Syncer) SyncCheckpoint(ctx context.Context, tsk types.TipSetKey) e
-		return xerrors.Errorf("failed to collect chain for checkpoint: %w", err)
-	}
+	if !hts.Equals(ts) {
+		if anc, err := syncer.store.IsAncestorOf(ctx, ts, hts); err != nil {
```
Might just be me but I find this style a lot more readable.

ZenGround0 commented on itests/kit/client.go:
```diff
@@ -141,7 +141,7 @@ func createRandomFile(rseed, size int) ([]byte, string, error) {
 	}
 	data := make([]byte, size)
-	rand.New(rand.NewSource(int64(rseed))).Read(data)
+	_, _ = rand.New(rand.NewSource(int64(rseed))).Read(data)
```
Maybe propagate this error ?

ZenGround0 commented on markets/dagstore/miner_api.go:
```diff
@@ -201,7 +201,7 @@ func (m *minerAPI) GetUnpaddedCARSize(ctx context.Context, pieceCid cid.Cid) (ui
+	l := pieceInfo.Deals[0].Length
 
-	return uint64(len), nil
+	return uint64(l), nil
```
Maybe remove the intermediate `l` and just `return uint64(pieceInfo.Deals[0].Length)`

ZenGround0 commented on markets/dagstore/miner_api_test.go:
```diff
@@ -129,9 +129,9 @@ func TestLotusAccessorGetUnpaddedCARSize(t *testing.T) {
 	// Check that the data length is correct
 	//stm: @MARKET_DAGSTORE_GET_UNPADDED_CAR_SIZE_001
-	len, err := api.GetUnpaddedCARSize(ctx, cid1)
+	l, err := api.GetUnpaddedCARSize(ctx, cid1)
```
Maybe `size` `l` is just too close to `1` for me

rvagg commented on node/impl/full/eth_utils.go:
```diff
@@ -410,7 +411,6 @@ func lookupEthAddress(addr address.Address, st *state.StateTree) (ethtypes.EthAd
 		return ethAddr, nil
 	}
 
-	// Otherwise, use the masked address.
```
good catch, I'd rather not lose it, https://github.com/filecoin-project/lotus/pull/12007
